pkg/net/libp2p/key.go

```diff
@@ -49,11 +50,16 @@ func operatorPublicKeyToNetworkPublicKey(
 		return nil, fmt.Errorf("libp2p supports only secp256k1 operator keys")
 	}
 
-	return &libp2pcrypto.Secp256k1PublicKey{
-		Curve: DefaultCurve,
-		X:     operatorPublicKey.X,
-		Y:     operatorPublicKey.Y,
-	}, nil
+	operatorPublicKeyBytes := operator.MarshalCompressed(operatorPublicKey)
+
+	btcecPublicKey, err := btcec.ParsePubKey(operatorPublicKeyBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	networkPublicKey := libp2pcrypto.Secp256k1PublicKey(*btcecPublicKey)
```

Comment:
@pdyraga @lukasz-zimnoch 
Done in 50761462a27df8068ceb0e7bbde92262f6123d00.

I used `UnmarshalSecp256k1PublicKey` and `UnmarshalSecp256k1PrivateKey` during conversions, but these functions return the `PubKey` and `PrivKey` interfaces instead of concrete types (`libp2pcrypto.Secp256k1PublicKey` and `libp2pcrypto.Secp256k1PrivateKey`).

Therefore I had to return `PubKey` and `PrivKey` from `operatorPublicKeyToNetworkPublicKey` and `operatorPrivateKeyToNetworkKeyPair` (an alternative would be an ugly down casting to concrete types before returning from these functions).

Notice that it forced me to downcast to concrete types in unit tests in `key_test.go`, as I need to have access to parameters of private/public keys (`x`, `y` and `D` values). An alternative here would be to marshal from `PubKey`/`PrivKey` to `btcec` keys.

Are you ok with these change?

Reply:
I think we should return concrete types as done previously. Even at the cost of the type assertion (there is no casting in Go). See my comments below.

Reply:
Returned the concrete types and added type assertions in 027febb4837f04ec11bc4e208c0b1e28a8da3642.